src/ucp/core/ucp_worker.c

```diff
@@ -2345,31 +2344,52 @@ static unsigned ucp_worker_discard_uct_ep_progress(void *arg)
     return 1;
 }
 
-static int ucp_worker_discard_remove_filter(const ucs_callbackq_elem_t *elem,
-                                            void *arg)
+static int
+ucp_worker_discard_remove_filter(const ucs_callbackq_elem_t *elem, void *arg)
 {
-    if ((elem->cb != ucp_worker_discard_uct_ep_destroy_progress) &&
-        (elem->cb != ucp_worker_discard_uct_ep_progress)) {
+    if ((elem->arg != arg) ||
```

Comment:
minor: to improve readability better to test arg in separate if-statement or invert to positive condition check

Reply:
done